File: 40-extra/07-exists-or-not.cpp
#include<iostream>
#include <unordered_map>
using namespace std;

int main(){
	int t,n,m,q;
	cin>>t;
	while(t--){
		unordered_map<int, bool> st;
		cin>>n;
		int arr[n];
		for(int i=0; i<n; i++){
			cin>>arr[i];
			st[arr[i]]=true;
		}
		cin>>m;
		for(int i=0; i<m;i++){
			cin>>q;
			if(st[q])
				cout<<"Yes\n";
			else
				cout<<"No\n";
		}


	}
	return 0;
}

File: 40-extra/07-exists-or-not.py
def tst():
	mapp = {}# int:bool
	input() # n
	arr = input().strip().split(' ')# list of str number
	nq = int(input().strip())

	for i in arr: #make hash
		mapp[i] = True

	for _ in range(nq):
		q = input().strip() # '1'
		if mapp.get(q):
			print("Yes")
		else:
			print("No")

def tst1():
    try:
    	mapp = {}# int:bool
    	n = int(input())
    	arr = list(map(int,input().split(' ')))
    	nq = int(input())
    
    	for i in arr: #make hash
    		mapp[i] = True
    
    	qarr = list(map(int, input().split(' ')))
    	for q in qarr:
    		if mapp.get(q):
    			print("Yes")
    		else:
    			print("No")
    except:
        print("Error")

t = int(input())
for _ in range(t):
	tst()
	#tst1()


